scipy/interpolate/interpolate.py

```diff
@@ -536,15 +536,24 @@ class _PPolyBase(object):
     """
     Base class for piecewise polynomials.
     """
-    __slots__ = ('c', 'x', 'extrapolate')
+    __slots__ = ('c', 'x', 'extrapolate', 'axis')
 
-    def __init__(self, c, x, extrapolate=None):
+    def __init__(self, c, x, extrapolate=None, axis=0):
         self.c = np.asarray(c)
         self.x = np.ascontiguousarray(x, dtype=np.float64)
         if extrapolate is None:
             extrapolate = True
         self.extrapolate = bool(extrapolate)
 
+        if not (0 <= axis < self.c.ndim):
+            raise ValueError("%s must be between 0 and %s" % (axis, c.ndim))
+
+        # roll the interpolation axis to be the first one in self.c
+        self.axis = axis
+        if axis != 0:
+            self.c = np.rollaxis(self.c, axis+1)
+            self.c = np.rollaxis(self.c, axis+1)
```

Comment:
Not really. The error is five lines above, in the validation: `axis == c.ndim-1` should not be allowed since `c` array needs two axes for coefficients and intervals. For `axis=0` the shape of c  is `(k, m, ...)`.

I've fixed the validation above and added a test.